src/flow/src/expr/scalar.rs

```diff
@@ -24,6 +24,22 @@ use snafu::ensure;
 use crate::adapter::error::{Error, InvalidQuerySnafu, UnsupportedTemporalFilterSnafu};
 use crate::expr::error::{EvalError, InvalidArgumentSnafu, OptimizeSnafu};
 use crate::expr::func::{BinaryFunc, UnaryFunc, UnmaterializableFunc, VariadicFunc};
+use crate::repr::ColumnType;
+
+/// A scalar expression with a known type.
+#[derive(Debug, Clone)]
+pub struct TypedExpr {
+    /// The expression.
+    pub expr: ScalarExpr,
+    /// The type of the expression.
+    pub typ: ColumnType,
```

Comment:
Why not `type`? 🤔

Reply:
`type` is a keyword, although we can use `r#type`

Reply:
While it's a common preference debt between `ty` and `typ`. Golang's internal code use `typ` while Rust's internal code use `ty` more commonly.